src/config/reader.rs

```diff
@@ -261,8 +261,8 @@ impl ConfigReader {
 
         while let Some(file) = queue.pop_front() {
             for import in file.dependency.iter() {
-                let proto = self.read_proto(import).await?;
                 if descriptors.get(import).is_none() {
+                    let proto = self.read_proto(import).await?;
```

Comment:
it used to unnecessarily read proto files.. it was just 1 line change so I kept it in this pr